python-lib/gluonts_forecasts/training_session.py

```diff
@@ -0,0 +1,257 @@
+import pandas as pd
+import os
+import math
+from pandas.api.types import is_numeric_dtype, is_string_dtype
+from gluonts_forecasts.model import Model
+from constants import METRICS_DATASET, METRICS_COLUMNS_DESCRIPTIONS, TIMESERIES_KEYS
+from gluonts_forecasts.gluon_dataset import GluonDataset
+from gluonts_forecasts.model_handler import list_available_models
+
+
+class TrainingSession:
+    """
+    Class to train and evaluate multiple GluonTS estimators on a training dataframe, and to retrieve an aggregated metrics dataframe
+
+    Attributes:
+        target_columns_names (list): List of column names to predict
+        time_column_name (str)
+        frequency (str): Pandas timeseries frequency (e.g. '3M')
+        epoch (int): Number of epochs used by the GluonTS Trainer class
+        models_parameters (dict): Dictionary of model names (key) and their parameters (value)
+        prediction_length (int): Number of time steps to predict
+        training_df (DataFrame): Training dataframe
+        make_forecasts (bool): True to output the evaluation predictions of the last prediction_length time steps
+        external_features_columns_names (list): List of columns with dynamic real features over time
+        timeseries_identifiers_names (list): Columns to identify multiple time series when data is in long format
+        batch_size (int): Size of batch used by the GluonTS Trainer class
+        num_batches_per_epoch (int): Number of batches per epoch
+        gpu (str): Not implemented
+        context_length (int): Number of time steps used by model to make predictions
+    """
+
+    def __init__(
+        self,
+        target_columns_names,
+        time_column_name,
+        frequency,
+        epoch,
+        models_parameters,
+        prediction_length,
+        training_df,
+        make_forecasts,
+        external_features_columns_names=None,
+        timeseries_identifiers_names=None,
+        batch_size=None,
+        user_num_batches_per_epoch=None,
+        gpu=None,
+        context_length=None,
+    ):
+        self.models_parameters = models_parameters
+        self.models = None
+        self.glutonts_dataset = None
+        self.training_df = training_df
+        self.prediction_length = prediction_length
+        self.target_columns_names = target_columns_names
+        self.time_column_name = time_column_name
+        self.frequency = frequency
+        self.epoch = epoch
+        self.make_forecasts = make_forecasts
+        self.external_features_columns_names = external_features_columns_names
+        self.use_external_features = len(external_features_columns_names) > 0
+        self.timeseries_identifiers_names = timeseries_identifiers_names
+        self.session_name = None
+        self.session_path = None
+        if self.make_forecasts:
+            self.forecasts_df = pd.DataFrame()
+            self.evaluation_forecasts_df = None
+        self.train_list_dataset = None
+        self.test_list_dataset = None
```

Comment:
After re-reading this module, I am wondering if this is the right name for this variable. If I understand correctly, this is not a "test" dataset but actually the full dataset. If that's indeed the case, a renaming would be welcome to clarify.

Reply:
It is the full dataset but also a test dataset for the evaluation part.
An accurate renaming could be:
- train_list_dataset => evaluation_train_list_dataset
- test_list_dataset => full_list_dataset

Reply:
good names!

Reply:
done in 648a0b4